webrecorder/browsertrix
# Compute name from seed URLs in UI

Makes config name field optional and enforces maximum, and renders name based on seed URL. Closes https://github.com/webrecorder/browsertrix-cloud/issues/574 and fixes https://github.com/webrecorder/browsertrix-cloud/issues/643.

### Manual testing
1. Log in and go to New Crawl Config
2. Choose URL List
3. Enter multiple URLs and save without entering a name. Verify config name is shown as `<first URL> +<remaining URL count> URLs`
4. Repeat for Seeded URL type. Verify config name is shown as `<primary URL>`
5. Run a crawl. Verify crawl name is shown the same as the config name

### Screenshots
**Crawl list of crawls without custom config name:**
<img width="1096" alt="Screen Shot 2023-02-27 at 12 29 19 PM" src="https://user-images.githubusercontent.com/4672952/221680452-693be741-f884-46af-80d8-0b21d12aeed3.png">

**Crawl list of URL list, hover demo:**


https://user-images.githubusercontent.com/4672952/221689999-22bf72a3-7ba7-46e7-801e-8cc4a7c01906.mov


**Crawl list of crawl with custom config name at 50 characters:**
<img width="1097" alt="Screen Shot 2023-02-27 at 12 41 08 PM" src="https://user-images.githubusercontent.com/4672952/221680847-2ac2ff64-fb0d-4f1e-9b12-9e191259d332.png">

**Crawl detail:**
<img width="1131" alt="Screen Shot 2023-02-27 at 12 35 01 PM" src="https://user-images.githubusercontent.com/4672952/221680550-a49dcbdc-6e3f-439d-abb6-70f33feb8847.png">

**Optional name field:**
<img width="880" alt="Screen Shot 2023-02-27 at 12 35 31 PM" src="https://user-images.githubusercontent.com/4672952/221680620-b1866ff7-070b-4dc1-a650-4cfc003947a4.png">

*Name field character limit:**
<img width="528" alt="Screen Shot 2023-02-27 at 12 35 45 PM" src="https://user-images.githubusercontent.com/4672952/221680655-55ff13b2-0c4e-4baa-8a83-8797953e19a9.png">

### Opinions
I couldn't find a good recommendation on max length for the name, so I chose `50` based on:
- roughly the number of characters that fit in the input field without scrolling
- some of the config names in dev.browsertrix.cloud
- the crawls list column width

Could be easily changed if there's a UX standard.

### Follow-ups

For folks currently using the unlimited-character name field to describe the config, this will limit descriptions they can attach to a config. I think it'd be valuable to add a longer `description` field on the crawl config object to address this.
